pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CastTransformFunction.java

```diff
@@ -130,6 +132,9 @@ public TransformResultMetadata getResultMetadata() {
             _stringValuesSV[i] = new Timestamp(longValuesSV[i]).toString();
           }
           break;
+        case JSON:
```

Comment:
No need to add this case here. `JSON` type doesn't require extra conversion

Reply:
Removed.